backend/services.endorsement-reminder/EndorsementMaintenanceService.cs

```diff
@@ -45,6 +49,14 @@ public async Task ExpireOldEndorsementRequestsAsync()
 
         foreach (var request in oldRequests)
         {
+            var email = request.Status == EndorsementRequestStatus.Approved
+                    ? request.RequestingParty!.Email!
```

Comment:
On previous projects you might be used to lazy loading, where the `RequestingParty` navigation property would be resolved by the database when accessed while unloaded. This causes non-obvious performance issues in situations like this, where the code would be hitting the DB once for every request we need to fetch the `Email` from.

In our project, entities are [eager loaded](https://learn.microsoft.com/en-us/ef/core/querying/related-data/eager). This means that if you don't `.Include()` or `.Select()` into the property it won't be loaded. This means that `RequestingParty` will be null, and this line will throw a `NullReferenceException`. The simplest solution would be to just add the `.Include()`:
``` C#
var oldRequests = await this.context.EndorsementRequests
    .Include(request => request.RequestingParty)
    .Where(request => inProgressStatuses.Contains(request.Status)
        && request.StatusDate < now - Duration.FromDays(30))
    .ToListAsync();
```

As an aside: whenever you use the null-forgiving operator `!`, you are telling the compiler "trust me, I'm confident that the value won't be null, and I'm willing to have the program throw an unhanded exception if I'm wrong". In this case, you should double-check that any navigation properties you use are being loaded by an `.Include()` or `.Select()` statement before squashing the warning that they might be null;